src/basic/hashmap.c

```diff
@@ -1853,10 +1853,13 @@ int _set_put_strdup_full(Set **s, const struct hash_ops *hash_ops, const char *p
         if (r < 0)
                 return r;
 
-        if (set_contains(*s, (char*) p))
-                return 0;
+        if (n == SIZE_MAX) {
+                if (set_contains(*s, (char*) p))
```

Comment:
why is this check done only for the n == SIZE_MAX case?

Reply:
When `n != SIZE_MAX`, to check the existence, we need to call `strndup()` first. But calling `set_contains()` then `set_consume()` is quite redundant. Hence, the check is done only when `n == SIZE_MAX`.

Or in other words, if and only if `n == SIZE_MAX`, we can check the existence without duplicating the string.